v4.1

Anarmard opened the pull request:
docker
- структура
- dockerfile / docker-compose

Anarmard committed:
v4.1
docker
- структура
- dockerfile / docker-compose

Anarmard committed:
v4.1
docker
- структура
- dockerfile / docker-compose

Anarmard committed:
v4.3
docker
BookingController

Anarmard committed:
v4.4
docker
ItemController

Anarmard committed:
v4.5
docker

Anarmard committed:
v4.6
docker
-fix pom

alexache commented on gateway/src/main/java/ru/practicum/shareitgateway/booking/BookingClient.java:
```diff
@@ -0,0 +1,68 @@
+
+    // Подтверждение или отклонение запроса на бронирование
+    public ResponseEntity<Object> approve(Long userId, Long bookingId, Boolean approved) {
+        Map<String, Object> parameters = Map.of(
```
Мапу из одного элемента можно убирать и делать простую подстановку в URI

alexache commented on gateway/src/main/java/ru/practicum/shareitgateway/booking/BookingController.java:
```diff
@@ -0,0 +1,99 @@
+    public ResponseEntity<Object> approveBooking(@RequestHeader(USERID) Long userId,
+                                             @PathVariable Long bookingId,
+                                             @RequestParam("approved") Boolean approved) {
+        return bookingClient.approve(userId, bookingId, approved);
```
Лучше логгирование оставить для всех методов контроллера

Anarmard replied:
Поправил, спасибо

alexache commented on gateway/src/main/java/ru/practicum/shareitgateway/item/ItemController.java:
```diff
@@ -0,0 +1,87 @@
+    @PostMapping
+    public ResponseEntity<Object> addItem(@RequestHeader(USERID) Long userId,
+                                      @Valid @RequestBody ItemCreateDto itemCreateDto) {
+        return itemClient.add(userId, itemCreateDto);
```
Логгирование

Anarmard replied:
Прошелся по всем контроллерам, поправил. Спасибо

alexache commented on pom.xml:
```diff
@@ -11,55 +11,64 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
-
```
Зачем все эти зависимости родительскому модулю? Может подразумевается dependencyManagement для выставление нужных версий? В остальном, у моделей разные зависимости

Anarmard replied:
Александр, pom наверное мое самое слабое место, которое я не понимаю(( Просто добавляю все зависимости, которые когда либо были у меня в проекте. Но в данном конкретном случае - это наверное сыграло со мной злую шутку: тесты не проходят возможно как раз из-за этого. 

Сейчас попробую удалить эти зависимости.

Anarmard replied:
Убрал еще валидацию из dto

alexache commented on server/src/main/java/ru/practicum/shareitserver/booking/controller/BookingController.java:
```diff
@@ -19,7 +19,7 @@ public class BookingController {
     @PostMapping
     public BookingResponseDto saveBooking(@RequestHeader(USERID) Long userId,
-                               @Valid @RequestBody BookingCreateRequestDto bookingCreateRequestDto) {
+                                          @Valid @RequestBody BookingCreateRequestDto bookingCreateRequestDto) {
```
Никаких валидаций не должно оставаться в модуле сервера. Из-за путаница с зависимостями родительского модуля это не так очевидно

Anarmard replied:
Да, вот сомневался удалять их или нет. 
Наверное надо удалить и другие проверки, которые у меня есть в BookingService модуль server?  Так как все эти проверки теперь есть в модуле gateway

alexache replied:
Все дублирующие проверки (и в целом проверки без использования базы) нужно убрать, верно

Anarmard committed:
v4.7
удалил из родительского пом зависимости

Anarmard committed:
v4.8
#---

Anarmard committed:
v4.8.1
#---
убрал валидацию из ДТО

Anarmard committed:
v4.8.2
#---
убрал валидацию из ДТО
забыл / в UserControler

Anarmard committed:
v4.8.3
#---
убрал валидацию из ДТО
забыл / в UserControler
post => patch

Anarmard committed:
v4.9
fix

Anarmard committed:
v4.9.1
fix

Anarmard committed:
v4.9.2
fix

Anarmard commented:
Александр, спасибо тебе огромное за подсказку!!! Удалил лишнее из родительского pom и ситуация исправилась.

Anarmard committed:
v4.9.3
remove some checks from Controller

Anarmard committed:
v4.9.4
при удалении проверок из Server => ServiceImpl валятся тесты, вернул проверки обратно
